### ISSUE TITLE
[Request] Add support for Nuget

### ISSUE THREAD

**Issue by lifeformed** [FlaxEngine/FlaxEngine#2074]
It would be great to be able to use nuget packages.  I'm trying to integrate the Amazon Gamelift SDK but it's quite a challenge without nuget.  It has a lot of system dependencies that aren't getting found.

**Comment by mafiesto4** [comment]
`Flax.Build` could expose additional light of nuget packages (with versions, etc.) to be used for project generation and build-time (https://docs.flaxengine.com/manual/editor/flax-build/index.html#building-binaries).

**Comment by lifeformed** [comment]
The dll I'm using references System.Configuration.ConfigurationManager, so I added it to the ScriptingAPI.SystemReferences, but it still couldn't find it. I've been trying to hunt down .NET dlls but it's been a pain to get the right versions and find the right places to put them.

**Comment by mafiesto4** [comment]
Yeah, it would be easier to use if we just implemented proper support in our build tool to handle all the stuff with dlls.

**Comment by mafiesto4** [comment]
Added in #3548. Docs: https://github.com/FlaxEngine/FlaxDocs/blob/c07614130e5f8af5985f51c238f1f04f328aec3f/manual/editor/flax-build/nuget-packages.md
